DeleteUserDataActivity/handler.ts

```diff
@@ -0,0 +1,617 @@
+ * @param param0.userDataBackup information about the blob storage account to place backup into
+ * @param param0.notification the notification
+ */
+const backupAndDeleteNotification = ({
+  notificationModel,
+  userDataBackup,
+  notification
+}: {
+  notificationModel: NotificationModel;
+  userDataBackup: IBlobServiceInfo;
+  notification: RetrievedNotification;
+}): TaskEither<
+  BlobCreationFailure | QueryFailure | DocumentDeleteFailure,
+  RetrievedNotification
+> => {
+  return sequenceT(taskEitherSeq)<
+    QueryFailure | BlobCreationFailure | DocumentDeleteFailure,
+    // tslint:disable-next-line: readonly-array
+    [
+      TaskEither<
+        QueryFailure | BlobCreationFailure | DocumentDeleteFailure,
+        RetrievedNotification
+      >,
+      TaskEither<
+        QueryFailure | BlobCreationFailure | DocumentDeleteFailure,
+        string
+      >
+    ]
+  >(
+    saveDataToBlob<RetrievedNotification>(
+      userDataBackup,
+      `notification--${notification.id}.json`,
+      notification
+    ),
+    fromQueryEither(() =>
+      notificationModel.deleteNotification(
+        notification.messageId,
+        notification.id
+      )
+    ).mapLeft(toDocumentDeleteFailure)
+  ).map(_ => notification);
+};
+
+/**
+ * Find all versions of a notification status, then backup and delete each document
+ * @param param0.notificationStatusModel instance of NotificationStatusModel
+ * @param param0.userDataBackup information about the blob storage account to place backup into
+ * @param param0.notification parent notification
+ *
+ */
+const backupAndDeleteNotificationStatus = ({
+  notificationStatusModel,
+  userDataBackup,
+  notification
+}: {
+  notificationStatusModel: NotificationStatusModel;
+  userDataBackup: IBlobServiceInfo;
+  notification: RetrievedNotification;
+}): TaskEither<
+  QueryFailure | BlobCreationFailure | DocumentDeleteFailure,
+  readonly RetrievedNotificationStatus[]
+> => {
+  return executeRecursiveBackupAndDelete<RetrievedNotificationStatus>(
+    item =>
+      notificationStatusModel.deleteNotificationStatusVersion(
+        item.notificationId,
+        item.id
+      ),
+    userDataBackup,
+    item => `notification-status--${item.version}.json`,
+    notificationStatusModel.findAllVersionsByModelId(notification.id)
+  );
+};
+
+/**
+ * Backup and delete a given message
+ *
+ * @param param0.messageStatusModel instance of MessageStatusModel
+ * @param param0.userDataBackup information about the blob storage account to place backup into
+ * @param param0.message the message
+ */
+const backupAndDeleteMessage = ({
+  messageModel,
+  userDataBackup,
+  message
+}: {
+  messageModel: MessageModel;
+  userDataBackup: IBlobServiceInfo;
+  message: RetrievedMessageWithoutContent;
+}): TaskEither<
+  BlobCreationFailure | QueryFailure | DocumentDeleteFailure,
+  RetrievedMessageWithoutContent
+> => {
+  return sequenceT(taskEitherSeq)<
+    QueryFailure | BlobCreationFailure | DocumentDeleteFailure,
```

Comment:
maybe it's worth to define a type alias for `QueryFailure | BlobCreationFailure | DocumentDeleteFailure`